File: EmployeeMgmt/EmployeeMgmt.WebApi/Controllers/EmployeeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Net.Http;
using System.Web.Http;

namespace EmployeeMgmt.WebApi.Controllers
{
    using EmployeeMgmt.WebApi.Models;
    using EmployeeMgmt.Domain.Entities;
    using EmployeeMgmt.Application.Contracts;

    [AllowAnonymous]
    [RoutePrefix("api/funcionario")]
    public class EmployeeController : ApiController
    {
        private IAppEmployee appEmployee;

        public EmployeeController(IAppEmployee appEmployee)
        {
            this.appEmployee = appEmployee;
        }

        [HttpPost]
        [Route("cadastrar")]
        public HttpResponseMessage Post(EmployeeModelCreate model)
        {
            try
            {
                if (ModelState.IsValid)
                {
                    Employee e = new Employee()
                    {
                        Name = model.Name,
                        Occupation = model.Occupation,
                        Email = model.Email,
                        Birth = model.Birth
                    };

                    appEmployee.Create(e);

                    return Request.CreateResponse(HttpStatusCode.OK, "Funcionário cadastrado com sucesso.");
                }
                else
                {
                    var errors = new List<string>();

                    foreach (var state in ModelState)
                    {
                        foreach (var e in state.Value.Errors)
                        {
                            errors.Add(e.ErrorMessage);
                        }
                    }
                    return Request.CreateResponse(HttpStatusCode.Forbidden, errors);
                }
            }
            catch (Exception e)
            {
                return Request.CreateResponse(HttpStatusCode.BadRequest, e.Message);
            }
        }

        [HttpPut]
        [Route("editar")]
        public HttpResponseMessage Put(EmployeeModelEdit model)
        {
            try
            {
                if (ModelState.IsValid)
                {
                    Employee e = appEmployee.GetById(model.Id);

                    if(e != null)
                    {
                        e.Name = model.Name;
                        e.Occupation = model.Occupation;
                        e.Email = model.Email;
                        e.Birth = model.Birth;

                        appEmployee.Update(e);

                        return Request.CreateResponse(HttpStatusCode.OK, "Funcionário atualizado com sucesso.");
                    }
                    else
                    {
                        throw new Exception("Erro. Funcionário não encontrado.");
                    }
                    
                }
                else
                {
                    var errors = new List<string>();

                    foreach (var state in ModelState)
                    {
                        foreach (var e in state.Value.Errors)
                        {
                            errors.Add(e.ErrorMessage);
                        }
                    }
                    return Request.CreateResponse(HttpStatusCode.Forbidden, errors);
                }
            }
            catch (Exception e)
            {
                return Request.CreateResponse(HttpStatusCode.BadRequest, e.Message);
            }
        }

        [HttpDelete]
        [Route("excluir")]
        public HttpResponseMessage Delete(int id)
        {
            try
            {
                Employee e = appEmployee.GetById(id);

                if(e != null)
                {
                    appEmployee.Delete(e);

                    return Request.CreateResponse(HttpStatusCode.OK, "Funcionário excluido com sucesso.");
                }
                else
                {
                    throw new Exception("Erro. Funcionário não encontrado.");
                }

            }
            catch (Exception e)
            {
                return Request.CreateResponse(HttpStatusCode.BadRequest, e.Message);
            }
        }

        [HttpGet]
        [Route("obter")]
        public HttpResponseMessage GetOne(int id)
        {
            try
            {
                Employee e = appEmployee.GetById(id);

                if(e != null)
                {
                    var model = new EmployeeModelGet()
                    {
                        Id = e.Id,
                        Name = e.Name,
                        Occupation = e.Occupation,
                        Email = e.Email,
                        Birth = e.Birth
                    };

                    return Request.CreateResponse(HttpStatusCode.OK);
                }
                else
                {
                    throw new Exception("Erro. Funcionário não encontrado.");
                }
            }
            catch (Exception e)
            {
                return Request.CreateResponse(HttpStatusCode.BadRequest, e.Message);
            }
        }

        [HttpGet]
        [Route("listar")]
        public HttpResponseMessage GetAll()
        {
            try
            {
                var list = new List<EmployeeModelGet>();

                foreach (Employee e in appEmployee.GetAll())
                {
                    var model = new EmployeeModelGet()
                    {
                        Name = e.Name,
                        Occupation = e.Occupation,
                        Email = e.Email,
                        Birth = e.Birth
                    };
                    list.Add(model);

                }
                return Request.CreateResponse(HttpStatusCode.OK);
            }
            catch (Exception e)
            {
                return Request.CreateResponse(HttpStatusCode.BadRequest, e.Message);
            }
        }
    }
}

File: EmployeeMgmt.Client/EmployeeMgmt.Web/app/controllers/employeeController.js
app.controller(
    'EmployeeCreate',
    function ($scope, $http) {

        $scope.title = "Página de cadastro de funcionários";

        $scope.model = {}; 

        $scope.create = function () {

            $scope.message = "Enviando requisição, por favor aguarde...";

            $http.post("http://localhost:3137/api/funcionario/cadastrar", $scope.model)
                .success(
                    function (msg) {
                        $scope.message = msg;
                        $scope.model = {}; 
                    }
                )
                .error(
                    function (e) {
                        $scope.message = e;
                    }
                );
        };
    }
);


app.controller(
    'EmployeeList',
    function ($scope, $http) {

        $scope.title = "Página de consulta de funcionários";

        $scope.model = {}; 

        $scope.getOne = function (id) {

            $http.get("http://localhost:3137/api/funcionario/obter?id=" + id)
                .success(
                    function (data) {
                        $scope.model.EmployeeId = data.EmployeeId;
                        $scope.model.Name = data.Name;
                        $scope.model.Occupation = data.Occupation;
                        $scope.model.Email = data.Email;
                        $scope.model.Birth = data.Birth;
                    }
                )
                .error(
                    function (e) {
                        $scope.message = e;
                    }
                );
        };

        $scope.edit = function () {

            $http.put("http://localhost:3137/api/funcionario/editar", $scope.model)
                .success(
                    function (msg) {
                        $scope.message = msg;
                        $scope.list();
                    }
                )
                .error(
                    function (e) {
                        $scope.message = e;
                    }
                );
        };

        $scope.delete = function (id) {

            $http.delete("http://localhost:3137/api/funcionario/excluir?id=" + id)
                .success(
                    function (msg) {
                        $scope.message = msg;
                        $scope.list();
                    }
                )
                .error(
                    function (e) {
                        $scope.message = e;
                    }
                );
        };

        $scope.getAll = function () {

            $http.get("http://localhost:3137/api/funcionario/listar")
                .success(
                    function (list) {
                        $scope.employees = list;
                    }
                )
                .error(
                    function (e) {
                        $scope.message = e; 
                    }
                );
        };
    }
);